src/main/java/org/betonquest/betonquest/objectives/CommandObjective.java

```diff
@@ -0,0 +1,76 @@
+package org.betonquest.betonquest.objectives;
+
+import org.betonquest.betonquest.BetonQuest;
+import org.betonquest.betonquest.Instruction;
+import org.betonquest.betonquest.api.Objective;
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.utils.PlayerConverter;
+import org.bukkit.Bukkit;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerCommandPreprocessEvent;
+
+import java.util.List;
+
+/**
+ * Requires the player to execute a specific command.
+ */
+@SuppressWarnings("PMD.CommentRequired")
+public class CommandObjective extends Objective implements Listener {
+
+    private final String command;
+    private final List<String> commandVariables;
+
+    public CommandObjective(final Instruction instruction) throws InstructionParseException {
+        super(instruction);
+        template = ObjectiveData.class;
+        command = instruction.next().replaceAll("_", " ");
+        commandVariables = BetonQuest.resolveVariables(command);
```

Comment:
This is not correct. This is not your fault, the way variables work is really unhandy (will be improved in BQ 2.0).

you may want to look here, how variables need to be resolved: https://github.com/BetonQuest/BetonQuest/blob/master/src/main/java/org/betonquest/betonquest/events/NotifyEvent.java#L79
I think you can simply "copy paste" this method.

Reply:
I took a look at `NotifyEvent` and it looks like it's doing a whole bunch of work. Is that because the variables extracted by `resolveVariables` are not actually _validated_ to be real variables? If so, then the code changes make sense.

Reply:
The problem is that you need to call the method `BetonQuest.createVariable(instruction.getPackage(), variable);`, otherwise it will not be resolved in `BetonQuest.getInstance().getVariableValue(instruction.getPackage().getName(), variable, playerID));`

It is simply really bad design. But we will reimplement variables completely in BQ 2.0. So just copy past it now

Reply:
It actually resolved just fine when we did testing earlier. It did not need the createVariable to be run to resolve %player%. Is that a special case?

Reply:
yes, if you use for example PlaceholderAPI variables, it should not work. Some for every variable with dynamic content with BQ.